### DISCUSSION TITLE
Detect just dates and adresses in a document

### DISCUSSION THREAD

**Discussion by paulgromy** [mindee/doctr#1275] [Q&A]
I want to learn use the KIE predictor.  I am using the code provided in the library.

from doctr.io import DocumentFile
from doctr.models import kie_predictor

model = kie_predictor(det_arch='db_resnet50', reco_arch='crnn_vgg16_bn', pretrained=True)
doc = DocumentFile.from_pdf("path/to/your/doc.pdf")
result = model(doc)

predictions = result.pages[0].predictions
for class_name in predictions.keys():
    list_predictions = predictions[class_name]
    for prediction in list_predictions:
        print(f"Prediction for {class_name}: {prediction}")

example my output

Prediction for words: Prediction(value='3174307256', confidence=0.6, bounding_box=((0.099609375, 0.0634765625), (0.1923828125, 0.0771484375)))
Prediction for words: Prediction(value='Phone:', confidence=1.0, bounding_box=((0.0458984375, 0.0634765625), (0.099609375, 0.078125)))
Prediction for words: Prediction(value='RS', confidence=1.0, bounding_box=((0.642578125, 0.0625), (0.6728515625, 0.078125)))
Prediction for words: Prediction(value='Trans#', confidence=1.0, bounding_box=((0.673828125, 0.0634765625), (0.736328125, 0.078125)))
Prediction for words: Prediction(value='2848419', confidence=1.0, bounding_box=((0.7392578125, 0.0634765625), (0.81640625, 0.078125)))
Prediction for words: Prediction(value='E', confidence=1.0, bounding_box=((0.076171875, 0.080078125), (0.091796875, 0.09375)))
Prediction for words: Prediction(value='486', confidence=0.99, bounding_box=((0.044921875, 0.080078125), (0.0771484375, 0.0947265625)))
Prediction for words: Prediction(value='Road', confidence=0.91, bounding_box=((0.1513671875, 0.080078125), (0.1923828125, 0.0947265625)))
Prediction for words: Prediction(value='Stop', confidence=1.0, bounding_box=((0.0908203125, 0.080078125), (0.1279296875, 0.095703125)))
Prediction for words: Prediction(value='18', confidence=0.98, bounding_box=((0.1279296875, 0.0791015625), (0.1513671875, 0.095703125)))
Prediction for words: Prediction(value='Generated:', confidence=1.0, bounding_box=((0.56640625, 0.0859375), (0.6640625, 0.099609375)))
Prediction for words: Prediction(value='Dec', confidence=1.0, bounding_box=((0.66796875, 0.0849609375), (0.7060546875, 0.1005859375)))
Prediction for words: Prediction(value='2022', confidence=1.0, bounding_box=((0.7275390625, 0.0859375), (0.7734375, 0.1005859375)))
Prediction for words: Prediction(value='2:20', confidence=1.0, bounding_box=((0.775390625, 0.0859375), (0.81640625, 0.1005859375)))
Prediction for words: Prediction(value='AM', confidence=1.0, bounding_box=((0.8173828125, 0.0849609375), (0.8505859375, 0.1005859375)))
Prediction for words: Prediction(value='46143', confidence=0.97, bounding_box=((0.1572265625, 0.09375), (0.2041015625, 0.10546875)))
Prediction for words: Prediction(value='Greenwood', confidence=1.0, bounding_box=((0.0458984375, 0.0927734375), (0.130859375, 0.1064453125)))

However, all words in a document are defined as the "words" class. But in my document there are phone numbers, addresses and the like. 

Can you tell me if I can set up the code in such a way that words are defined as the "words" class (at the moment everything is defined by this class), addresses are defined as the "addresses" class, dates as the "dates" class.

The description of the documentation says that there is such a possibility. How to implement it?

**Comment by felixdittrich92** [comment]
Hi @paulgromy 👋,

The base kie detection model provides only the 'words' label (same as the standard detection). 
You need to fine tune the model on your own custom dataset :)

**Reply by felixdittrich92** [reply]
https://github.com/mindee/doctr/tree/main/references/detection#data-format
